src/PSRule.Rules.Azure/rules/Azure.ACR.Rule.ps1

```diff
@@ -48,4 +48,13 @@ Rule 'Azure.ACR.GeoReplica' -Ref 'AZR-000004' -Type 'Microsoft.ContainerRegistry
     return $Assert.Fail($LocalizedData.ReplicaNotFound);
 }
 
+# Synopsis: Azure Container Registries should have soft delete policy enabled and retention period to 90 days.
+Rule 'Azure.ACR.SoftDelete' -Ref 'AZR-000299' -Type 'Microsoft.ContainerRegistry/registries' -Tag @{ release = 'Preview'; ruleSet = '2022_09'; } {
+    $notGeoReplica = @(GetSubResources -ResourceType 'Microsoft.ContainerRegistry/registries/replications')
+    $Assert.Count($notGeoReplica, '.', 0).Reason($LocalizedData.ACRNotGeoReplica, $TargetObject.name)
```

Comment:
@BernieWhite 

I'm little lost here and need some help. Also reason for that I uploaded the drafted product.

What is the best method definition to use to ensure that the array is empty? Looks like all the methods requires the field parameter in the overload defintions. All I want to do is to ensure that the array is of count 0. I found this example used in another rule with the use of `'.'`, that I suppose reflect the object itself? So object, object and count if I'm correct?

Reply:
@BernieWhite 

I think you didn't see this question from my side, thanks in advance.

Reply:
@bengeset96 Yes `.` refers to the object itself which is passed into the first parameter.